articles/creating-a-hangman-game-with-vanilla-javascript/index.md

```diff
@@ -0,0 +1,238 @@
+```javascript
+function checkIfGameWon() {
+  if (wordStatus === answer) {
+    document.getElementById('keyboard').innerHTML = 'You Won!!!';
+  }
+}
+```
+
+###checkIfGameLost() function
+Here’s the code to check if the game lost.
+```javascript
+function checkIfGameLost() {
+  if (mistakes === maxWrong) {
+    document.getElementById('wordSpotlight').innerHTML = 'The answer was: ' + answer;
+    document.getElementById('keyboard').innerHTML = 'You Lost!!!';
+  }
+}
+```
+In the code above, We set the game to print out “You lost!!!” when the player has reached the maximum number of wrong guesses. We also set the game to print out the correct answer too.
+reset() function
+In this function, we are going to set everything back to default.  Here is how it looks in the code:
+```javascript
+function reset() {
+  mistakes = 0;
+  guessed = [];
+  document.getElementById('hangmanPic').src = 'images/0.jpg';
+
+  randomWord();
+  guessedWord();
+  updateMistakes();
+  generateButtons();
+}
+```
+###updateHangmanPicture () function
+Firstly, we are going to run the `updateHangmanPicture();` in the `elseif` statement of the `handleGuess()` function so that the image updates every time the player gets a letter wrong. Then we create the updateHangmanPicture() function using the code below:
+```javascript
+function updateHangmanPicture() {
+  document.getElementById('hangmanPic').src = 'images/' + mistakes + '.jpg';
+}
+```
+In the code above, we set the images to update each time the player gets an answer wrong continually. I named all six images (0-6).jpg, and we have a maximum number of 6 mistakes to be made. This now conveniently helps us set the pictures to update every time the player gets a letter wrong.
+
+Lastly, we need to make the game accessible to players that wish to navigate with their keyboard. Firstly, you need to define the alphabet keys in your keyboard using the code below:
+```javascript
+Let alphabets=["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"];
+```
+Next, you need to create a function that will link these keys on your keyboard to the game and make them appear as valid guesses as they are clicked by using `document.onkeypress`. Here is the code below:
+```javascript
+document.onkeypress = function (e) {
+  e = e || window.event;
+  var charCode = e.charCode || e.keyCode,
+      character = String.fromCharCode(charCode);
+
+  if (alphabets.includes(character))
+  handleGuess(character);
+};
+```
+So as to say “in the event of a key being pressed, handle it as a guess in the game.”
+Below is a video of the game in action with both the mouse and keyboard in use:
+
+`<iframe width="478" height="269" src="https://www.youtube.com/embed/WBThhqHnn2I" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>`
```

Comment:
If this Iframe is supposed to be embedded in your article, please remove it from the code blocks.